modules/kmm-customizing-upgrades-for-kernel-modules.adoc

```diff
@@ -13,39 +13,53 @@ Use this procedure to upgrade the kernel module while running maintenance operat
 
+
 .Prerequisites
 
-* Before upgrading, set the `kmm.node.kubernetes.io/version-module.<module-namespace>.<module-name>=$moduleVersion` label on all the nodes that will be used by the kernel module.
+* Before upgrading, set the `kmm.node.kubernetes.io/version-module.<module_namespace>.<module_name>=$moduleVersion` label on all the nodes that are used by the kernel module.
 
 * Terminate all user application workloads on the node or move them to another node.
 
 * Unload the currently loaded kernel module.
 
 * Ensure that the user workload (the application running in the cluster that is accessing kernel module) is not running on the node prior to kernel module unloading and that the workload is back running on the node after the new kernel module version has been loaded.
 
-* Ensure that the device plugin managed by KMM on the node is unloaded before the upgrade process and reloaded following the upgrade process.
-
-* Due to Kubernetes limitations in label names, the combined length of `Module` name and namespace must not exceed 39 characters.
-
 .Procedure
 
-. Update the following fields in the `Module` CR:
+. Ensure that the device plugin managed by KMM on the node is unloaded.
+
+. Update the following fields in the `Module` custom resource (CR):
 - `containerImage` (to the appropriate kernel version)
 - `version`
 +
 The update should be atomic; that is, both the `containerImage` and `version` fields must be updated simultaneously.
 
 . Terminate any workload using the kernel module on the node being upgraded.
 
-. Remove the `kmm.node.kubernetes.io/version-module.<module-namespace>.<module-name>` label on the node. This unloads the kernel module from the node.
-
-. If required, perform any additional maintenance required on the node for the kernel module upgrade.
-
-. Add the `kmm.node.kubernetes.io/version-module.<module-namespace>.<module-name>=$moduleVersion` label to the node. `$moduleVersion` should be equal to the new value of the `version` field in the `Module` CR.
+. Remove the `kmm.node.kubernetes.io/version-module.<module_namespace>.<module_name>` label on the node.
+Run the following command to unload the kernel module from the node:
++
+[source,terminal]
+----
+$ oc label node/<node_name> kmm.node.kubernetes.io/version-module.<module_namespace>.<module_name>-
+----
 
-. Restore any workload that leverages the kernel module on the node.
+. If required, as the cluster administrator, perform any additional maintenance required on the node for the kernel module upgrade.
++
+If no additional upgrading is needed, you can skip Steps 3 through 6 by updating the `kmm.node.kubernetes.io/version-module.<module-namespace>.<module-name>` label value to the new `$moduleVersion` as set in the `Module`.
 
+. Run the following command to add the `kmm.node.kubernetes.io/version-module.<module_namespace>.<module_name>=$moduleVersion` label to the node. `$moduleVersion` should be equal to the new value of the `version` field in the `Module` CR.
```

Comment:
```suggestion
. Run the following command to add the `kmm.node.kubernetes.io/version-module.<module_namespace>.<module_name>=$moduleVersion` label to the node. The `$moduleVersion` must be equal to the new value of the `version` field in the `Module` CR.
```

Reply:
Changes made.

Reply:
You're in luck! I was just trawling my GitHub inbox and saw this as soon as it popped up. Merging meow!